backend/src/services/course/course.service.ts

```diff
@@ -71,6 +83,20 @@ class Course {
       }
     }
   }
+
+  async getByCategoryName(
```

Comment:
We need to get category by key, not by name

Reply:
Also we would have at least two filtering options for courses (by name and by category keys) so we need to combine it in `getAll` courses method - so I think you should wait for Farid's changes :c